src/main/java/com/avispl/symphony/dal/communicator/barco/e2gen2/BarcoE2Communicator.java

```diff
@@ -1059,7 +1184,15 @@ private void clearLayerFromDest(int currentScreenDestId, int excludeLayerId, boo
 			if (isMixedType && currentLayerID == excludeLayerId + 1) {
 				continue;
 			}
-			layerRequestDTOList.add(new LayerRequestDTO(currentLayerID));
+			int linkDestId = screenDestContent.get(i).get(BarcoE2Constant.LINK_DEST_ID).asInt();
```

Comment:
screenDestContent.get(i) can be extracted and reused

Reply:
Fixed